bigquery_etl/parse_udf.py

```diff
@@ -155,12 +178,32 @@ def udf_usage_definitions(text, raw_udfs=None):
     ]
 
 
+def udf_tests_sql(raw_udf, raw_udfs):
+    """
+    Create tests for testing persistent UDFs.
+
+    Persistent UDFs need to be rewritten as temporary UDF so that changes
+    can be tested.
+    """
+    tests_full_sql = []
+    for test in raw_udf.tests:
+        test_sql = prepend_udf_usage_definitions(test, raw_udfs)
+        test_sql = sub_persistent_udfs_as_temp(test_sql)
+        test_sql = test_sql.replace(
+            "CREATE OR REPLACE FUNCTION", "CREATE TEMP FUNCTION"
+        )
```

Comment:
this should be case insensitive, match any whitespace, and cover all variations of persistent udf definitions (`CREATE FUNCTION`, `CREATE OR REPLACE FUNCTION`, `CREATE FUNCTION IF NOT EXISTS`).

for example:
```python
UDF_DEFINITION_KEYWORD = re.compile(r"CREATE\s+(OR\s+REPLACE\s+)?FUNCTION(\s+IF\s+NOT\s+EXISTS)?", re.IGNORECASE)

...

        test_sql = UDF_DEFINITION_KEYWORD.sub("CREATE TEMP FUNCTION", test_sql)
```